src/main/java/org/jenkinsci/plugins/vSphereCloudSlaveTemplate.java

```diff
@@ -645,15 +647,10 @@ public FormValidation doTestCloneParameters(@AncestorInPath AbstractFolder<?> co
         @Nonnull
         @Restricted(NoExternalUse.class) // used by Jelly EL only
         public List<NodePropertyDescriptor> getNodePropertiesDescriptors() {
-            List<NodePropertyDescriptor> result = new ArrayList<NodePropertyDescriptor>();
-            final Jenkins j = Jenkins.getInstance();
-            final List<NodePropertyDescriptor> list = j.getDescriptorList(NodeProperty.class);
-            for (NodePropertyDescriptor npd : list) {
-                if (npd.isApplicable(vSphereCloudSlave.class)) {
-                    result.add(npd);
-                }
-            }
-            return result;
+            DescriptorExtensionList<NodeProperty<?>, NodePropertyDescriptor> nodePropertyDescriptors = NodeProperty.all();
+            return nodePropertyDescriptors.stream()
+                .filter(npd -> npd.isApplicable(vSphereCloudSlave.class))
+                .collect(Collectors.toList());
```

Comment:
I'm not hugely keen of the aesthetics on the syntax here but I presume that this is Java-8 syntax for applying the same filter?
(I'm not asking for changes, just reassurances that this is "how it's normally done these days" as I've not seen that code pattern before)

Reply:
you can ask IDEA to switch stream to foreach it will look exactly the same.

Reply:
Switched to foreach